doc/requirements/Requirements.md

```diff
@@ -0,0 +1,704 @@
+<article class="markdown-body">
+
+High level requirements
+=======================
+
+<nb>Some requirements are only motivated by the fact that the reference
+implementation implements them. Search for "reference implementation".</nb>
+
+# Introduction
+The PF **MUST** be a library.
+<why>To be reused in different components.</why>
+
+PF instances **MUST NOT** share any mutable data.
+<why>to guarantee that different PF instances will not impact each others.</why>
+
+# Parameters
+
+A PF **MUST** be able to handle parameters.
+<why>because the PF aims to abstract hardware and model it by parameters.</why>
+
+## Definitions
+
+<dl>
+<dt>Hardware</dt>
+<dd>System controlled by the PF. Not necessary material system. This term was
+chosen because:
+
+ - historically the PF reference implementation was used to abstract hardware
+ - the subsystem term would arguably fit best is already used.
+
+(FIXME: choose subsystem instead, it is not used in fact.)
+</dd>
+</dl>
+
+## Value
+
+A parameter **MUST** have a value.
+<why>because a parameter without value would not abstract any hardware.</why>
+
+A PF **MUST** support mutable parameters.
+<why>To control the underlined hardware.</why>
+
+A PF **MAY** support immutable parameters, i.e. parameters which value is determined
+on start then read only.
+<why>To permit hardware read only value reflection.</why>
+<ko>This is not implemented in the PF reference implementation.</ko>
+
+This value **MUST** be gettable and settable (except if immutable).
+<why>A parameter that can not be accessed is of no use.</why>
+
+A value **MUST** be convertible to a string.
+<why>To display a parameter value to the human user for debugging purpose.</why>
```

Comment:
quite strange requirement... I presonnally dislike libs that log stuff. I just want the lib to work, I don't want to debug it.